### PR TITLE
Extend output formats of `repo alias list` command

### PR DESCRIPTION
**Pull request by demusdev**
Allow output in all the supported formats.

## Description

Closes: #751

## Checklist before requesting a review

- [ ] Unit and integration tests added
  
- [ ] Compatibility:
    
  - [ ] Network APIs: ✅
    
  - [ ] Workspace layout and metadata: ✅
    
  - [ ] Configuration: ✅
    
  - [ ] Container images: ✅
- [ ] Observability:
    
  - [ ] Tracing / [metrics](https://github.com/kamu-data/kamu-standards/blob/master/metrics_design.md): ✅
    
  - [ ] Alerts: ✅
- [ ] Documentation:
    
  - [ ] [Changelog](./CHANGELOG.md): ✅
    
  - [ ] [Public documentation](https://github.com/kamu-data/kamu-docs/): ✅
  
- [ ] Downstream effects:
    
  - [ ] [kamu-node](https://github.com/kamu-data/kamu-node): ✅
    
  - [ ] [kamu-web-ui](https://github.com/kamu-data/kamu-web-ui): ✅
    
  - [ ] [release-queue](https://github.com/orgs/kamu-data/projects/4/views/1)

### CODE REVIEW

**Review by s373r — COMMENTED** [review]
Added few minor comments

File: CHANGELOG.md [hunk 1]
```diff
@@ -22,6 +22,8 @@ Recommendation: for ease of reading, use the following order:
  - Broken catalog issue for server and transactional modes  
 
 ## [0.204.3] - 2024-09-26
+### Changed
+- `kamu repo alias list` now supports JSON alonside with other formats mentioned in the command's help
```
[the remark this answers is not in the document]
Comment (demusdev) [reply CHANGELOG.md]: Thanks for catching this.

File: src/app/cli/src/commands/alias_list_command.rs [hunk 2]
```diff
@@ -39,38 +41,32 @@ impl AliasListCommand {
...
+            Field::new("Alias", DataType::Utf8, false),
+        ]))
+    }
 
-        table.set_titles(row![bc->"Dataset", bc->"Kind", bc->"Alias"]);
+    async fn records(
+        &self,
+        schema: Arc<Schema>,
+        datasets: &Vec<DatasetHandle>,
+    ) -> Result<RecordBatch, GetAliasesError> {
```
[the remark this answers is not in the document]
Comment (demusdev) [reply src/app/cli/src/commands/alias_list_command.rs:66]: Make sense. thanks!

**Comment by sergiimk** [comment]
Superseded by #870 where I have rebased and squashed the changes and added one more small tweak.

Thanks for the contribution @demusdev!

**Pull request closed without merging**